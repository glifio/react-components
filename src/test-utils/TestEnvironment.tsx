import { ReactNode } from 'react'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { CoinType } from '@glif/filecoin-address'
import LotusRPCEngine from '@glif/filecoin-rpc-client'
import { Environment, Network, networks } from '../services/EnvironmentProvider'

const OptionalApolloMockProvider = ({ children, mocks, withApollo }) => {
  if (withApollo || mocks.length > 0)
    return <MockedProvider mocks={mocks}>{children}</MockedProvider>

  return <>{children}</>
}

export const TestEnvironment = ({
  children,
  withApollo,
  apolloMocks
}: TestEnvironmentProps) => (
  <Environment
    coinType={CoinType.TEST}
    networkName={Network.CALIBRATION}
    nodeStatusUrl={networks[Network.CALIBRATION].nodeStatusUrl}
    graphUrl={networks[Network.CALIBRATION].graphUrl}
    graphSecure={networks[Network.CALIBRATION].graphSecure}
    lotusApiUrl={networks[Network.CALIBRATION].lotusApiUrl}
    explorerUrl='https://explorer-calibration.glif.link'
    lotusApi={
      new LotusRPCEngine({
        apiAddress: networks[Network.CALIBRATION].lotusApiUrl
      })
    }
  >
    <OptionalApolloMockProvider withApollo={withApollo} mocks={apolloMocks}>
      {children}
    </OptionalApolloMockProvider>
  </Environment>
)

type TestEnvironmentProps = {
  children: ReactNode
  withApollo?: boolean
  apolloMocks?: MockedResponse[]
}

TestEnvironment.defaultProps = {
  withApollo: false,
  apolloMocks: []
}
