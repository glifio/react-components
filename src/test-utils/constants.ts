export const WALLET_ADDRESS = 't1z225tguggx4onbauimqvxzutopzdr2m4s6z6wgi'
export const WALLET_ID = 't0101'

export const WALLET_ADDRESS_2 = 't1nq5k2mps5umtebdovlyo7y6a3ywc7u4tobtuo3a'
export const WALLET_ID_2 = 't0102'

export const WALLET_ADDRESS_3 = 't1mbk7q6gm4rjlndfqw6f2vkfgqotres3fgicb2uq'
export const WALLET_ID_3 = 't0103'
