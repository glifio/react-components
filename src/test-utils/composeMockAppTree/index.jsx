/* eslint-disable react/prop-types */
import React from 'react'
import { mockWalletProviderInstance } from '../../../__mocks__/@glif/filecoin-wallet-provider'
import WalletProviderWrapper from '../../services/WalletProvider'
import { initialState as walletProviderInitialState } from '../../services/WalletProvider/state'
import { composeWalletProviderState } from './composeState'
import { TestEnvironment } from '../TestEnvironment'

jest.mock('../../services/WalletProvider')

const Index = (statePreset = 'preOnboard', options = {}) => {
  // here you can pass a walletProviderInitialState and a preset to shape the store how you want it for testing
  const initialState = composeWalletProviderState(
    options?.walletProviderInitialState || walletProviderInitialState,
    statePreset
  )

  let walletProviderCache = { ...initialState }

  const cacheWalletProviderState = state => {
    walletProviderCache = { ...state }
    return <></>
  }

  const getWalletProviderState = () => walletProviderCache

  const Tree = ({ children }) => {
    return (
      <TestEnvironment>
        <WalletProviderWrapper
          options={options}
          statePreset={statePreset}
          getState={cacheWalletProviderState}
          initialState={initialState}
        >
          {children}
        </WalletProviderWrapper>
      </TestEnvironment>
    )
  }

  return {
    Tree,
    walletProvider: mockWalletProviderInstance,
    getWalletProviderState
  }
}

/**
 * This function is a wrapper that mocks everything the filecoin app needs for testing
 *
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * walletProviderDispatch(a dispatcher to mock/read calls to the walletProvider's state)
 * pathname
 * query
 */

export default Index
