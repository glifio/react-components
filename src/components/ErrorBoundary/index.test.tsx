import { render } from '@testing-library/react'
import { TestEnvironment } from '../../test-utils/TestEnvironment'

import { ErrorBoundary } from '.'

const ThrowError = () => {
  throw new Error('Test')
}

describe('ErrorBoundary', () => {
  // silence the thrown error for this test
  beforeAll(() => {
    jest.spyOn(console, 'error').mockImplementation(() => {})
  })

  test('it should catch errors and render the error boundary component', () => {
    const { container, getByText } = render(
      <TestEnvironment>
        <ErrorBoundary>
          <ThrowError />
        </ErrorBoundary>
      </TestEnvironment>
    )

    expect(getByText(/Glif is currently down/)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
