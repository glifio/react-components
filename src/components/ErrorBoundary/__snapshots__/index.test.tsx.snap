// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary it should catch errors and render the error boundary component 1`] = `
.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  gap: 0.75rem;
  padding: 0.5em 0.75em;
  color: var(--black);
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  border-color: var(--black);
  background: transparent;
}

.c7 svg path {
  stroke: var(--black);
}

.c7:hover {
  color: var(--purple-medium);
  border-color: var(--purple-medium);
}

.c7:hover svg path {
  stroke: var(--purple-medium);
}

.c7:active {
  color: var(--white) !important;
  border-color: var(--purple-medium) !important;
  background: var(--purple-medium) !important;
}

.c7:active svg path {
  stroke: var(--white) !important;
}

.c5 {
  vertical-align: middle;
}

.c3 {
  padding: 1.5em;
  border-radius: 8px;
  overflow: hidden;
  text-align: center;
  word-break: break-word;
}

.c3 > *:first-child {
  margin-top: 0;
}

.c3 > *:last-child {
  margin-bottom: 0;
}

.c3 > svg:first-child {
  display: block;
  margin: 0 auto 1em;
}

.c3 > h2:first-child {
  margin: 0;
}

.c3 header:first-child {
  margin: -1.5em -1.5em 1.5em -1.5em;
  padding: 1.5em;
  background-color: var(--purple-medium);
  color: var(--white);
}

.c4 {
  background-color: var(--red-light);
  color: var(--red-dark);
}

.c4 hr {
  border-color: var(--red-dark);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: var(--space-m);
  margin: 1.5em 0;
}

.c0 {
  position: relative;
  margin: 3rem 1.5rem;
}

.c0 > *:first-child {
  margin-top: 0;
}

.c0 > *:last-child {
  margin-bottom: 0;
}

.c1 {
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--space-m);
  width: 100%;
  margin: 0 auto;
  max-width: 40em;
}

.c2 form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--space-m);
}

@media (min-width:868px) {
  .c0 {
    margin: 3rem;
  }
}

<div
  class="c0 c1"
>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <svg
        class="c5 has-stroke"
        fill="none"
        height="20"
        stroke="var(--red-dark)"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 22 20"
        width="22"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 7.74982V9.99986M11 14.4999H11.0112M3.24282 19H18.7572C20.4815 19 21.5586 17.1246 20.6965 15.6249L12.9393 2.12474C12.0771 0.625088 9.92286 0.625088 9.0607 2.12474L1.30352 15.6249C0.441363 17.1246 1.5185 19 3.24282 19Z"
        />
      </svg>
      <h2>
        Glif is currently down
      </h2>
      <hr />
      <p>
        We've been notified of the outage and expect to be back up and running again shortly.
      </p>
      <hr />
      <p>
        Get help in 
        <a
          href="https://discord.gg/B9ju5Eu4Rq"
          rel="noreferrer noopener"
          target="_blank"
        >
          Discord
        </a>
         or hit us up on 
        <a
          href="https://twitter.com/glifio"
          rel="noreferrer noopener"
          target="_blank"
        >
          Twitter
        </a>
      </p>
    </div>
    <div
      class="c6"
    >
      <button
        class="c7"
      >
        Back
      </button>
    </div>
  </div>
</div>
`;
